CMakeLists.txt

```diff
@@ -310,104 +310,35 @@ if(BML_INTERNAL_GEMM)
-  CACHE STRING "If set, the preferred BLAS/LAPACK vendor. Possible choices: {Intel,MKL,ACML,GNU,IBM}")
-
-if(BLAS_VENDOR STREQUAL "Intel" OR BLAS_VENDOR STREQUAL "MKL")
-  set(MKLROOT $ENV{MKLROOT})
-  message(STATUS "Attempting to use Intel's BLAS/LAPACK (MKL)")
-  if(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
-    message(STATUS "Look for MKL in ${MKLROOT}")
-    if(BML_OPENMP)
-      set(BLAS_LIBRARIES "-L${MKLROOT}/lib/intel64"
-        -lmkl_lapack95_lp64 -lmkl_intel_lp64 -lmkl_core -lmkl_intel_thread
-        -lpthread -lm)
-    else()
-      set(BLAS_LIBRARIES "-L${MKLROOT}/lib/intel64"
-        -lmkl_intel_lp64 -lmkl_core -lmkl_sequential -lpthread -lm)
-    endif()
-  else()
-    if(BML_OPENMP)
-      set(BLAS_LIBRARIES "-L${MKLROOT}/lib/intel64"
-        -Wl,--no-as-needed -lmkl_gf_lp64 -lmkl_gnu_thread -lmkl_core -lgomp
-        -lpthread -lm -ldl)
-    else()
-      set(BLAS_LIBRARIES -Wl,--no-as-needed "-L${MKLROOT}/lib/intel64"
-        -lmkl_lapack95_lp64 -lmkl_gf_lp64 -lmkl_sequential -lmkl_core
-        -lmkl_sequential -lmkl_core -lpthread -lm)
-    endif()
-  endif()
-  set(BLAS_FOUND TRUE)
-  set(LAPACK_FOUND TRUE)
-elseif(BLAS_VENDOR STREQUAL "ACML")
-  if(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
-    if(BML_OPENMP)
-      set(BLAS_LIBRARIES "-L$ENV{ACML_DIR}/ifort64_mp/lib" -lacml_mp)
-    else()
-      set(BLAS_LIBRARIES "-L$ENV{ACML_DIR}/ifort64/lib" -lacml)
-    endif()
-  elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "GNU"
-      OR CMAKE_Fortran_COMPILER_ID STREQUAL "PathScale")
-    if(BML_OPENMP)
-      set(BLAS_LIBRARIES "-L$ENV{ACML_DIR}/gfortran64_mp/lib" -lacml_mp)
-    else()
-      set(BLAS_LIBRARIES "-L$ENV{ACML_DIR}/gfortran64/lib" -lacml)
-    endif()
-  elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "PGI")
-    if(OPENMP)
-      set(BLAS_LIBRARIES "-L$ENV{ACML_DIR}/pgi64_mp/lib" -lacml_mp)
-    else()
-      set(BLAS_LIBRARIES "-L$ENV{ACML_DIR}/pgi64/lib" -lacml)
-    endif()
-  endif()
-  set(BLAS_FOUND TRUE)
-  set(LAPACK_FOUND TRUE)
-elseif(BLAS_VENDOR STREQUAL "GNU")
-  if(CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
-    set(BLAS_LIBRARIES  -llapack -lblas)
-  endif()
-  set(BLAS_FOUND TRUE)
-  set(LAPACK_FOUND TRUE)
-elseif(BLAS_VENDOR STREQUAL "IBM")
-  if(CMAKE_Fortran_COMPILER_ID STREQUAL "XL")
-    set(BLAS_LIBRARIES "-L$ENV{ESSL_DIR}/lib64" -lesslsmp)
-  endif()
-  set(BLAS_FOUND TRUE)
-  set(LAPACK_FOUND TRUE)
-elseif(BLAS_VENDOR STREQUAL "None")
+  CACHE STRING "If set, the preferred BLAS/LAPACK vendor. Possible choices: {OpenBLAS,Intel,ACML,IBMESSL,...}")
+
+if(BLAS_VENDOR STREQUAL "None")
   set(NOBLAS TRUE)
   add_definitions(-DNOBLAS)
-elseif(BLAS_VENDOR STREQUAL "Auto")
-  set(BLAS_FOUND TRUE)
-  set(LAPACK_FOUND TRUE)
-  message(STATUS "I will skip BLAS/LAPACK detection")
-elseif(BLAS_VENDOR STREQUAL "")
-  message(STATUS "I will rely on the default paths or environment variables to find BLAS/LAPACK")
-else()
-  message(FATAL_ERROR "Unknown BLAS_VENDOR ${BLAS_VENDOR}")
+elseif(NOT BLAS_LIBRARIES)
```

Comment:
BLAS_LIBRARIES could have been set by the user through the build.sh script. In that case, it uses what the user is specifying instead of letting cmake find it

Reply:
Yes, you are right.